### DISCUSSION TITLE
Redis Client Pool - How does the redis connection pool work?

### DISCUSSION THREAD

**Discussion by gspaiva** [quarkusio/quarkus#33334] [Q&A]
Hey Quarkus team, I got a question.

I'm setting up a quarkus application that uses the quarkus redis-client library, I would like to have a connection pool with ready connections to be used. Although, when starting the application I don't see ready connections being opened in the Redis server, the number of connections opened only changes when I request the endpoint that interacts with Redis. I'm using the `ReactiveRedisDataSource` and `RedisDataSource` to interact with Redis.

I was expecting that the client would open the number of connections I configured via the `quarkus.redis.max-pool-size` property, however, it seems like it's opening them lazily. Although that's not much of a problem, I checked that once I'm done sending commands to Redis for the request, the connection is closed. I connected to a Redis instance directly and saw that although I made a lot of requests to my endpoint, the Redis connection number always dropped afterward, which means the connections need to be reopened on every request. Isn't that a performance issue?

For checking the number of active clients in the Redis server, I used the `redis-cli info` command.

Thanks in advance. If this is not the right place to post these questions, let me know.

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
The connections are created on demand (lazily) and recycled.

**Comment by cescoffier** [comment]
BTW, we you want to keep the same connection, use `.withConnection`.
